core/src/util/tileID.h

```diff
@@ -16,6 +16,7 @@ struct TileID {
     
     TileID(int _x, int _y, int _z) : x(_x), y(_y), z(_z) {};
     TileID(const TileID& _rhs): x(_rhs.x), y(_rhs.y), z(_rhs.z) {};
+    TileID(const TileID&& _rhs): x(std::move(_rhs.x)), y(std::move(_rhs.y)), z(std::move(_rhs.z)) {};
```

Comment:
I agree on we should not be using move on TileID "everywhere" but in cases where TileID is not needed in the original callee (example: https://github.com/tangrams/tangram-es/blob/datasource-cleanup/core/src/tile/tileManager.cpp#L38 and https://github.com/tangrams/tangram-es/blob/datasource-cleanup/core/src/tile/tileManager.cpp#L45), you can get the rvalue of the temporary object and move it to be used inside TileTask. 
To me it is more readable and yes nothing to do with performance.

Also one thing wrong about the move constructor of TileID is the use of "const T&&". Based on my going through multiple stackoverflow posts (one of these being: http://stackoverflow.com/questions/6307526/c0x-const-rvalue-reference-as-function-parameter), it is making sure a "const rvalue" parameter makes sure it is only bindable to an rvalue and nothing else!! So changing that move constructor to a non const rvalue is better. I could be interpreting this wrong though, so forgive me if I am wrong here :D. Based on a mutual agreement here, we can change the signature of TileID's move constructor.

@hjanetzek http://www.cprogramming.com/c++11/rvalue-references-and-move-semantics-in-c++11.html this describes why this kind of move is legal! (Read the section on "Detecting temporary objects with rvalue references).

Also on a similar but different topic, I also recommend reading the following on rvalue reference and universal reference (when to use std::move and when to use std::forward).. https://isocpp.org/blog/2012/11/universal-references-in-c11-scott-meyers and also http://scottmeyers.blogspot.com/2012/10/parameter-types-in-constructors.html (debate on whether to pass parameters to a constructor by value or const reference)